misk-crypto/src/main/kotlin/misk/crypto/pgp/RealPgpEncoder.kt

```diff
@@ -0,0 +1,133 @@
+package misk.crypto.pgp
+
+import org.bouncycastle.bcpg.ArmoredOutputStream
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.bouncycastle.openpgp.PGPCompressedData
+import org.bouncycastle.openpgp.PGPCompressedDataGenerator
+import org.bouncycastle.openpgp.PGPEncryptedData
+import org.bouncycastle.openpgp.PGPEncryptedDataGenerator
+import org.bouncycastle.openpgp.PGPEncryptedDataList
+import org.bouncycastle.openpgp.PGPException
+import org.bouncycastle.openpgp.PGPLiteralData
+import org.bouncycastle.openpgp.PGPLiteralDataGenerator
+import org.bouncycastle.openpgp.PGPOnePassSignatureList
+import org.bouncycastle.openpgp.PGPPrivateKey
+import org.bouncycastle.openpgp.PGPPublicKey
+import org.bouncycastle.openpgp.PGPPublicKeyEncryptedData
+import org.bouncycastle.openpgp.PGPUtil
+import org.bouncycastle.openpgp.jcajce.JcaPGPObjectFactory
+import org.bouncycastle.openpgp.operator.jcajce.JcePGPDataEncryptorBuilder
+import org.bouncycastle.openpgp.operator.jcajce.JcePublicKeyDataDecryptorFactoryBuilder
+import org.bouncycastle.openpgp.operator.jcajce.JcePublicKeyKeyEncryptionMethodGenerator
+import org.bouncycastle.util.io.Streams
+import java.io.ByteArrayOutputStream
+import java.security.SecureRandom
+import java.security.Security
+import java.util.Date
+
+internal class RealPgpEncoder(
+  private val privateKey: PGPPrivateKey
+) : PgpEncoder {
+  init {
+    Security.addProvider(BouncyCastleProvider())
+  }
+
+  override fun encrypt(
+    byteArray: ByteArray,
+    encryptionKey: PGPPublicKey,
+    armored: Boolean
+  ): ByteArray {
+    val buffer = okio.Buffer()
+
+    val byteArrayOutputStream = ByteArrayOutputStream()
+    val pgpCompressedDataGenerator = PGPCompressedDataGenerator(PGPCompressedData.ZIP)
+    val pgpLiteralDataGenerator = PGPLiteralDataGenerator()
+    val pgpLiteralDataStream = pgpLiteralDataGenerator.open(
+        pgpCompressedDataGenerator.open(byteArrayOutputStream),
+        PGPLiteralData.BINARY,
+        "", // Name of the file encoded in the data. Just boring metadata.
+        byteArray.size.toLong(),
+        Date()
+    )
+
+    pgpLiteralDataStream.write(byteArray)
+    pgpLiteralDataStream.close()
+    pgpCompressedDataGenerator.close()
+
+    val outputStream = buffer.outputStream()
+    val out = when (armored) {
+      true -> ArmoredOutputStream(outputStream)
+      else -> outputStream
+    }
+
+    val jcePGPDataEncryptorBuilder = JcePGPDataEncryptorBuilder(PGPEncryptedData.CAST5)
+        .setWithIntegrityPacket(true)
```

Comment:
What does this do?

Reply:
"Sets whether or not the resulting encrypted data will be protected using an integrity packet."

From https://tools.ietf.org/html/rfc4880#section-5.13: "An implementation MUST support decrypting these packets and SHOULD prefer generating them to the older Symmetrically Encrypted Data packet when possible."